plugins/filterFactory.go

```diff
@@ -145,6 +145,15 @@ func makeVolumeFilterConfig(configInput string) (*VolumeFilterConfig, error) {
 		return nil, fmt.Errorf("invalid input (%s), the third part needs to be \"base\" or \"quote\"", configInput)
 	}
 
+	// marketIDs and account IDs might not be present in the config string, but cannot be nil on the config
+	if config.additionalMarketIDs == nil {
```

Comment:
@debnil this is significantly reducing the quality of the codebase. now we have two places that "set" these two arrays, here and in `parseVolumeFilterModifier`. please remove.

I'm not sure if we had discussed adding this logic, but if there was a previous comment that indicated it please let me know so I can clarify requested changes better.

Reply:
These lines changed because now, when we validate the config, we check that market IDs and account IDs are not nil. This function is what creates a config. Previously, when parsing a config string, if it did not specify

The point of this pull request is to validate the config. That requires making changes like this to other functions that are now invalid. We could in theory split these in a separate PR, but that seems meaningless without the validation changes. If these changes are a bad idea, then we should either reconsider the validation scheme or change how the parsing logic works above this - I think this approach does that in the cleanest way possible, given the way that market and account IDs are parsed.

Reply:
Sounds good - these are good clarifications, changes made!